src/HubSpotServiceProvider.php

```diff
@@ -3,32 +3,59 @@
 namespace Rossjcooper\LaravelHubSpot;
 
 use Illuminate\Support\ServiceProvider;
+use SevenShores\Hubspot\Delay;
+use SevenShores\Hubspot\RetryMiddlewareFactory;
 
 class HubSpotServiceProvider extends ServiceProvider
 {
-	/**
-	 * Register the service provider.
-	 */
-	public function register()
-	{
-		//Bind the HubSpot wrapper class
-		$this->app->bind('Rossjcooper\LaravelHubSpot\HubSpot', function ($app) {
-			return HubSpot::create(
-				env('HUBSPOT_API_KEY', config('hubspot.api_key')),
-				null,
-				config('hubspot.client_options', [])
-			);
-		});
-	}
-
-	/**
-	 * Perform post-registration booting of services.
-	 */
-	public function boot()
-	{
-		// config
-		$this->publishes([
-			__DIR__.'/config/hubspot.php' => config_path('hubspot.php'),
-		], 'config');
-	}
+    /**
+     * Register the service provider.
+     */
+    public function register()
+    {
+        //Bind the HubSpot wrapper class
+        $this->app->bind('Rossjcooper\LaravelHubSpot\HubSpot', function ($app) {
+            $handlerStack = \GuzzleHttp\HandlerStack::create();
+            $handlerStack->push(
+                RetryMiddlewareFactory::createRateLimitMiddleware(
+                    Delay::getConstantDelayFunction()
+                )
+            );
+
+            $handlerStack->push(
+                RetryMiddlewareFactory::createInternalErrorsMiddleware(
+                    Delay::getExponentialDelayFunction(2)
+                )
+            );
+
+            $guzzleClient = new \GuzzleHttp\Client(['handler' => $handlerStack]);
+
+            // BC fix
+            if (!config('hubspot.key')) {

```

Comment:
Hi, I'm not sure if this bc fix works (tried it locally). I think it might be a better solution to have both `key` and `api_key` in the config that both reference the `HUBSPOT_API_KEY` env value.
Either that or building a config array to pass to the `Client` using the original hubspot config e.g. 

```php
$clientConfig = [
    'key' => config('hubspot.api_key),
    ...
];
```

Reply:
Yeah that makes sense.

I'm actually not using hubspot for anything anymore - was too limiting and doesn't really meet our needs.
I still think this PR - with your changes - would be a good thing to merge though - Hubspot has so many failed responses that retries then work.

cheers.